metadata/src/validation.rs

```diff
@@ -404,6 +495,19 @@ pub fn get_metadata_hash(metadata: &RuntimeMetadataV15) -> [u8; 32] {
         &mut visited_ids,
     ));
 
+    let mut apis: Vec<_> = metadata
```

Comment:
This is ok for now, but thoughts:

- we set the capacity of bytes above; can we now adjust that to take into account these extra things we're adding to them?
- more generally I wonder whether we can get rid of the allocations entirely; iirc if you XOR a set of `[u8;32]`'s together for instance, it doesnt matter what order they are `xor`'d in and they will output the same result. So maybe we don't need to be sorting at all.
- need to think carefully about `xor` use in other places though because of this; `concat_and_hash` is generally "safer" and all of our XOR-ing might lead to a bunch of stuff that'll collide more. (ie I think we should maybe be xor-ing when we don't care about order of inputs and concat_and_hash-ing when we do, eg if we end up with nameA,input1A,input2A,nameB,input1B,input2B then we should not just xor all of these together because it'll be identical to if we swapped the inputs around, but we could `xor(concat_and_hash(nameA,input1A,input2A), concat_and_hash(nameB,input1B,input2B))` for instance to say "we care about those internal valeus being hashed uniquely but dont care about the order of those named things

Let's open an issue to look into this all!

Reply:
Referencing the issue: https://github.com/paritytech/subxt/issues/937, thanks for opening it!